xbmc/guilib/guiinfo/GUIInfo.h

```diff
@@ -71,20 +71,23 @@ class CGUIInfo
   {
   }
 
+  CGUIInfo(int info, const std::string& data3, const std::string& data5)
+    : m_info(info), m_data1(0), m_data3(data3), m_data4(0), m_data5(data5)
+  {
+  }
+
   bool operator ==(const CGUIInfo &right) const
   {
-    return (m_info == right.m_info &&
-            m_data1 == right.m_data1 &&
-            m_data2 == right.m_data2 &&
-            m_data3 == right.m_data3 &&
-            m_data4 == right.m_data4);
+    return (m_info == right.m_info && m_data1 == right.m_data1 && m_data2 == right.m_data2 &&
+            m_data3 == right.m_data3 && m_data4 == right.m_data4 && m_data5 == right.m_data5);
   }
 
   uint32_t GetInfoFlag() const;
   uint32_t GetData1() const;
   int GetData2() const { return m_data2; }
   const std::string& GetData3() const { return m_data3; }
   int GetData4() const { return m_data4; }
+  const std::string& GetData5() const { return m_data5; }
```

Comment:
By the time we add our fifth anonymous variable name, we ask if this is becoming a dumping ground. Are there other approaches?

Reply:
Yeah it's definitely a dumping ground unfortunately, but I don't see other way to do it atm. We need to check globally the max number of data fields that are used at the same time. My guess is that only 2 (or 3 in the worst case scenario) are used simultaneously. We might be able to turn that into std::variants in the future but last time I tried to used osx failed (so it has to wait).
Any change here that is not as simple as this one will not only blow the diff but also create high risk...

Reply:
Yeah, agreed. I have no problem with data5 but it's a good time to start thinking of future scalability.